src/NuGet.Services.Validation.Orchestrator/Job.cs

```
@@ -190,14 +190,21 @@ private void ConfigureJobServices(IServiceCollection services, IConfigurationRoo
             {
                 var smtpConfigurationAccessor = serviceProvider.GetRequiredService<IOptionsSnapshot<SmtpConfiguration>>();
                 var smtpConfiguration = smtpConfigurationAccessor.Value;
+                if (string.IsNullOrWhiteSpace(smtpConfiguration.SmtpUri))
+                {
+                    return new MailSenderConfiguration();
```

Should this throw, so we fail fast?

---

It is not supposed to fail in this case. It will fallback to writing emails to disk if `Host` property is empty.

---

I wonder if we should fail instead... 

If we fail here, we'd be alerted on Orchestrator queue size growing and can quickly redeploy. If we continue and write emails to disk, would we not notice until we run out of disk space?

---

I'd guess it'd take years to run out of disk space saving emails.

---

What will be the e2e experience in the failure case?

---

Packages will be validated just fine, exception will be thrown after updating the database, but before deleting the package from the validation container, so we'll end up with some packages kept in validation container. I'd like to point out that this is the *current* behavior. It's just made a bit more explicit with this change. Previously we'd just have `Host` property in `MailSenderConfiguration` silently set to empty string. Now we need explicit check because we can't instantiate `Uri` class with empty string.